auth/src/main/java/com/android/swingmusic/auth/data/workmanager/ScheduleTokenRefreshWork.kt

```diff
@@ -0,0 +1,33 @@
+package com.android.swingmusic.auth.data.workmanager
+
+import android.content.Context
+import androidx.work.BackoffPolicy
+import androidx.work.Constraints
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
+import java.util.concurrent.TimeUnit
+
+fun scheduleTokenRefreshWork(context: Context) {
+    val constraints = Constraints.Builder()
+        .setRequiredNetworkType(NetworkType.CONNECTED)
+        .setRequiresBatteryNotLow(true)
+        .build()
+
+    val tokenRefreshWorkRequest = PeriodicWorkRequestBuilder<TokenRefreshWorker>(21, TimeUnit.DAYS)
+        .setConstraints(constraints)
+        .setBackoffCriteria(
+            BackoffPolicy.EXPONENTIAL,
+            1,
+            TimeUnit.HOURS
+        )
```

Comment:
This task is scheduled to run every 21 days — ideally a week before the token expires. If constraints are not met, the task is retried hourly with exponential backoff.

The delay formula is:

`delay = initial delay × (2 ^ retry attempt)`